CoorpAcademy/mobile
# Feat/modal on dead end

## Detailed purpose of the PR

The PR covers several dead end cases  : 

**### CASE 1 :** I try to launch the app even if I have no access to it (platform disabled).

--> Tech : we can detect when a platform is disabled if the server returns 403 request header
--> Product : as a user I must see a modal explaining to me that I can ask my administrtor to (re)activate the mobile APP.

The modal looks like this :

<img width="364" alt="Screenshot 2019-05-13 at 17 43 08" src="https://user-images.githubusercontent.com/15958334/57635353-43438f00-75a7-11e9-92c4-b8f3160f1ea1.png">

**HOW TO TEST IT ?**

In order to test, try to launch the application with a platform that does not have access to the application (eg : TagHeure). 

**### CASE 2 :** I try to launch the app but I have no recommendations, most popular nor other item to display on my dashboard

--> Tech : before desplaying the dashboard we try to fetch in // the most popular and the recommendations. After those API calls, if no cards were found, we should display the modal. The modal also show a refresh button. When triggering the button the last `redux action` will be call once more.

--> Product : as a user I must see a modal explaining that something happend and I can try to refresh the application by triggering a button

The modal looks like this : 

<img width="348" alt="Screenshot 2019-05-13 at 17 59 38" src="https://user-images.githubusercontent.com/15958334/57636165-e9dc5f80-75a8-11e9-898c-3e6ca8241bdd.png">

**HOW TO TEST IT ?**

To do so, go grap a test phone, dowload the PR 289 application version on your phone and log on by scanning the QR code available on `error-modal-staging.coorpacademy.com` platform.

Launch the app, you should see the accurate modal with the refresh button. You could click on it and see the application trying to refresh.

WARNING : no content should be shown because no catalog item is provided on the platform.




       If checked, what have you broken ?

- [X] **Extra dependency ?**
      If checked, Which extra dependency did you add ? (name, purpose, link ...). 
React Native Modal (https://github.com/react-native-community/react-native-modal) : a great modal handler library

### Testing Strategy

- [ ] Already covered by tests
